### PR TITLE
[DO NOT MERGE] make a Go-specific updater image

### PR DESCRIPTION
**Pull request by jakecoffman**
Doing some exploratory work to get an ecosystem-specific updater image started.

Starting with a lot of copy-pasting, don't be alarmed. 

Putting up a Draft PR to see what the CI thinks about all this.

### CODE REVIEW

File: Dockerfile.go_modules [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+COPY --chown=dependabot:dependabot .rubocop.yml ${DEPENDABOT_HOME}/.rubocop.yml
+
+WORKDIR ${DEPENDABOT_HOME}
+COPY --chown=dependabot:dependabot omnibus ${DEPENDABOT_HOME}/omnibus
+COPY --chown=dependabot:dependabot git_submodules ${DEPENDABOT_HOME}/git_submodules
+COPY --chown=dependabot:dependabot terraform ${DEPENDABOT_HOME}/terraform
+COPY --chown=dependabot:dependabot github_actions ${DEPENDABOT_HOME}/github_actions
+COPY --chown=dependabot:dependabot hex ${DEPENDABOT_HOME}/hex
+COPY --chown=dependabot:dependabot elm ${DEPENDABOT_HOME}/elm
+COPY --chown=dependabot:dependabot docker ${DEPENDABOT_HOME}/docker
```
Comment (honeyankit) [inline Dockerfile.go_modules:153]: Why we are copying all the other ecoysystem?

Comment (honeyankit) [reply Dockerfile.go_modules:153]: Nevermind this is an updater image.

Comment (jakecoffman) [reply Dockerfile.go_modules:153]: Yeah later if we add conditional imports we could not do the copy, but really doesn't take up much space to leave it.

File: .github/workflows/go-smoke.yml [hunk 2]
```diff
@@ -0,0 +1,86 @@
+# Runs all ecosystems cached and concurrently.
```
Comment (jeffwidman) [inline .github/workflows/go-smoke.yml:1]: the phrasing "cached and concurrently" is weird... can you reword to clarify?

File: Dockerfile.go_modules [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+    bison \
+    tzdata \
+    zip \
+    unzip \
+    locales \
+    openssh-client \
+    software-properties-common \
+    # Everything from here onwards is only installed to ensure
+    # Python support works with all packages (which may require
+    # specific libraries at install time).
```
Comment (jeffwidman) [inline Dockerfile.go_modules:39]: why not drop all the python libs?

or rather, why not start from a fresh / empty build and then add packages as needed?

File: Dockerfile.go_modules [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+  && mkdir -p /opt && chown dependabot:dependabot /opt && chgrp dependabot /etc/ssl/certs && chmod g+w /etc/ssl/certs
+
+
+### RUBY
+
+# When bumping Ruby minor, need to also add the previous version to `bundler/helpers/v{1,2}/monkey_patches/definition_ruby_version_patch.rb`
+ARG RUBY_VERSION=3.1.2
+ARG RUBY_INSTALL_VERSION=0.8.5
+
+ARG BUNDLER_V1_VERSION=1.17.3
```
Comment (jeffwidman) [inline Dockerfile.go_modules:76]: We don't need Bundler v1 for this...

File: Dockerfile.go_modules [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+
+WORKDIR $DEPENDABOT_HOME/dependabot-updater
+
+RUN bundle config set --local path 'vendor' && \
+bundle config set --local frozen 'true' && \
+bundle config set --local without 'development' && \
+bundle install
+
+
+# START: HACKY WORKAROUND FOR NPM GIT INSTALLS SPAWNING CHILD PROCESS
```
Comment (jeffwidman) [inline Dockerfile.go_modules:174]: is this necessary? since we don't use npm/yarn for go 😆

File: Dockerfile.go_modules [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+ENV NODE_EXTRA_CA_CERTS=/etc/ssl/certs/ca-certificates.crt
+
+# END: HACKY WORKAROUND FOR NPM GIT INSTALLS SPAWNING CHILD PROCESS
+
+# Add project
+COPY --chown=dependabot:dependabot updater /home/dependabot/dependabot-updater
+
+# Fix for git vulnerability since we run as root
+# see https://github.blog/2022-04-12-git-security-vulnerability-announced/
+RUN git config --global --add safe.directory /home/dependabot/dependabot-updater/repo
```
Comment (jeffwidman) [inline Dockerfile.go_modules:193]: this was removed too in a recent PR

**Review by jeffwidman — COMMENTED** [review]
saw you requested reviews, but it looks like it's not mergable... did you just want high-level feedback or on something specific?

**Comment by jakecoffman** [comment]
Changes have been merged into https://github.com/dependabot/dependabot-core/pull/6415

**Pull request closed without merging**